src/core/components/files.js

```diff
@@ -65,6 +65,14 @@ module.exports = function files (self) {
         return callback(new Error('Invalid arguments, data must be an object, Buffer or readable stream'))
       }
 
+      let total = 0
+      let prog = options.progress || (() => {})
+      const progress = (bytes) => {
+        total += bytes
+        prog(total)
+      }
+
+      options.progress = progress
       pull(
         pull.values(normalizeContent(data)),
         importer(self._ipldResolver, options),
```

Comment:
@diasdavid @VictorBjelkholm I noticed that there are two entry points for importer, here and above on line 26. The one above, gets called from the HTTP API, this one is called from the regular callback flow, they seem to be be doing exactly the same thing, perhaps we can replace this with a call to `createAddPullStream` and consolidate the progress accumulator logic there?